### PR TITLE
Fix reordering items logic in attributes/options lists

### LINKED ISSUE #36255: Reordering items in attributes/options lists breaks the other list
### Describe the bug

There is a problem when reordering items in attributes/options lists. Reordering items in one list removes every item in the other list.

Screen Capture on 2023-01-04 at 16-10-34

### Expected behavior

Reordering items in one list shouldn't affect the other list.

### Steps to reproduce

1. Add some attributes and options to a product.
2. Reorder the attributes.
3. Go to the `Options` section and verify that the list is empty now.

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR fixes the logic behind the attributes and options reordering. There was a bug that broke one list when the other one was reordered. This PR fixes that bug.

Before
Screen Capture on 2023-01-04 at 16-10-34

After
Screen Capture on 2023-01-04 at 16-12-50

Closes #36255.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Add some attributes and options to a product.
2. Reorder the attributes.
3. Go to the `Options` section and verify that the list is working well.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/fields/attribute-field/utils.ts [hunk 1]
```diff
@@ -16,27 +16,28 @@ export function getAttributeKey(
...
-	items: JSX.Element[],
+	items: Record< number | string, number >,
 	attributeKeyValues: Record< number | string, ProductAttribute >
 ): ProductAttribute[] {
-	return items
-		.map( ( { props }, index ): ProductAttribute | undefined => {
-			const key = getAttributeKey( props?.attribute );
-			if ( attributeKeyValues[ key ] ) {
+	return Object.keys( attributeKeyValues ).map(
+		( objectKeys: number | string ): ProductAttribute => {
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/fields/attribute-field/utils.ts]: I think this is the singular key, right?

```suggestion
		( attributeKey: number | string ): ProductAttribute => {
```

**Review by joshuatf — DISMISSED** [review]
Added one minor naming nitpick, but this is testing well and otherwise LGTM!

File: plugins/woocommerce-admin/client/products/fields/attribute-field/utils.ts (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce-admin/client/products/fields/attribute-field/utils.ts]: Good idea!

**Comment by octaedro** [comment]
Hey @joshuatf, I just renamed the const.
Could you take another look at this PR?

**Review by joshuatf — APPROVED** [review]
Looks good- thanks for renaming! :shipit:

**Pull request merged**